### PR TITLE
Doxygen improvements for sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90

### PR DESCRIPTION
**Pull request by lgannoaa**
Doxygen update for regional_esg_grid.fd/psym2.f90

Part of #366

### CODE REVIEW

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 [hunk 1]
```diff
@@ -101,109 +104,105 @@ subroutine eigensym2d(em,vv,oo,vvd,ood,ff)!                        [eigensym2]
...
 z(1,1)=em(2,2); z(2,1)=-em(2,1); z(1,2)=-em(1,2); z(2,2)=em(1,1)
 detem=em(1,1)*em(2,2)-em(2,1)*em(1,2)
 z=z/detem
 end subroutine invsym2
-!=============================================================================
+
+!> Get the inverse, z,of a 2*2 symmetric matrix, em, and its derivative, zd,
+!! with respect to symmetric variations of its components. I.e., for a
+!! symmetric infinitesimal change, delta_em, in em, the resulting
+!1 infinitesimal change in z would be:
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:124]: !1 should be !!

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 [hunk 2]
```diff
@@ -101,109 +104,105 @@ subroutine eigensym2d(em,vv,oo,vvd,ood,ff)!                        [eigensym2]
...
 call invsym2(em,z)
 call id2222(zd)
 do l=1,2; do k=1,2
    zd(:,:,k,l)=-matmul(matmul(z,zd(:,:,k,l)),z)
 enddo;    enddo
 end subroutine invsym2d
 
-!=============================================================================
+!> Get the sqrt of a symmetric positive-definite 2*2 matrix
+!! @author R. J. Purser
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:144]: Please put the author tag at the bottom of the docblock in all cases.

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 (hunk 1 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90]: Done

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 [hunk 3]
```diff
@@ -1,5 +1,6 @@
 !> @file
-!! @author R. J. Purser @date September 2018
+!! @author R. J. Purser 
+!! @date September 2018
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:3]: We need a blank line here, so there are two doxygen docblocks, one for the file, one for the module.

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 [hunk 4]
```diff
@@ -35,18 +36,17 @@ module psym2
 
 contains
 
-!=============================================================================
+!> Get the orthogonal eigenvectors, vv, and diagonal matrix of eigenvalues, oo,
+!! of the symmetric 2*2 matrix, em.
+!! @author R. J. Purser
```
Comment (edwardhartnett) [inline sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:41]: Move author tag to bottom in all cases.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 (hunk 2 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:144]: > Please put the author tag at the bottom of the docblock in all cases.

@edwardhartnett Please explain a bit more in detail on this comment. 
I inserted @author tag in subroutine sqrtsym2. What else should I do?

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 (hunk 4 above)
Comment (lgannoaa) [reply sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:41]: > Move author tag to bottom in all cases.

Would you please explain where should the @author tag located. I do not understand your comment - Move author tag to bottom in all cases.
You may use grid_tools.fd/regional_esg_grid.fd/psym2.f90 as an example.

File: sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90 (hunk 2 above)
Comment (edwardhartnett) [reply sorc/grid_tools.fd/regional_esg_grid.fd/psym2.f90:144]: If there is a date tag, I usually put it on the same line as the author:

`!! @author Freddy Fortran @date 12-Dec-1981
`
(The format of the date doesn't matter, can be 12/12/81, or just December 1981, or any reasonable date format.)

If there is no date in the existing documentaiton, I don't worry about it.

If there is no author, I look in the GitHub "blame" screen to show who checked in the code, or I just assume whoever wrote the file also wrote all the functions in the file.

**Review by edwardhartnett — APPROVED** [review]

**Pull request merged**